File: Python/invoice/get_invoice_by_id.py
from pagarmecoreapi.pagarmecoreapi_client import PagarmecoreapiClient
from pagarmecoreapi.controllers import *
from pagarmecoreapi.exceptions.error_exception import *

# Configuration parameters and credentials
basic_auth_user_name = 'basic_auth_user_name' # The username to use with basic authentication
basic_auth_password = 'basic_auth_password' # The password to use with basic authentication

PagarmecoreapiClient(basic_auth_user_name, basic_auth_password)


invoices_controller = invoices_controller.InvoicesController()

invoice_Id = "in_DKRdGqpsaVS4rmpl"

try:
    result = invoices_controller.get_invoice(invoice_Id)
    assert result is not None
    assert result.id is not None
    print("Invoice id: ", result.id)
    print("Invoice status: ", result.status)
except ErrorException as ex:
    print(ex.message)
    print("Errors: ", ex.errors)
except Exception as ex:
    raise ex
    

File: Java/invoice/GetInvoiceById.java
package examples.invoice;

import me.pagar.api.PagarmeCoreApiClient;
import me.pagar.api.controllers.InvoicesController;
import me.pagar.api.http.client.*;
import me.pagar.api.models.GetInvoiceResponse;

public class GetInvoiceById {

    public static void main(String[] args) {

        // Configuration parameters and credentials
        String basicAuthUserName = "basicAuthUserName"; // The username to use with basic authentication
        String basicAuthPassword = "basicAuthPassword"; // The password to use with basic authentication

        PagarmeCoreApiClient client = new PagarmeCoreApiClient(basicAuthUserName, basicAuthPassword);

        String invoiceId = "in_DKRdGqpsaVS4rmpl";

        InvoicesController invoices_controller = new InvoicesController();

        invoices_controller.getInvoiceAsync(invoiceId, new APICallBack<GetInvoiceResponse>() {
            @Override
            public void onSuccess(HttpContext context, GetInvoiceResponse response) {

                System.out.println("Invoice found!");
                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println("Charge ID: " + response.getId());

            }

            @Override
            public void onFailure(HttpContext context, Throwable error) {

                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println(error.getMessage());
                error.printStackTrace();

            }
        });
    }
}